src/darwin/Framework/CHIP/MTRDiagnosticLogsDelegate.h

```diff
@@ -0,0 +1,73 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRCluster.h>
+#import <Matter/MTRDefines.h>
+
+@class MTRDeviceController;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * The protocol definition for the MTRDiagnosticLogsDelegate.
+ *
+ */
+@protocol MTRDiagnosticLogsDelegate <NSObject>
+/**
+ * Notify the delegate when a BDX Session starts for some logs.
+ * The controller identifies the fabric the node is on, and the
+ * nodeID identifies the node within that fabric.
+ *
+ * If completion is passed a non-nil error, that will be converted into
+ * an error response to the sender. Otherwise a success response will be sent.
```

Comment:
```suggestion
 * an error response to the BDX initiator. Otherwise a success response will be sent.
```

Reply:
I will do the Darwin bits into a separated PR with a different API. Marking this one as resolved.